pems-server/src/main/java/org/maroubra/pemsserver/api/resources/SensorsResource.java

```diff
@@ -77,12 +79,13 @@ public void updateSensor(@PathParam("id") String id, @ApiParam(name = "JSON body
     }
 
     @GET
-    @Path("{id}/history")
+    @Path("{id}/{dataSize}/history")
     @ApiOperation(value = "Get a sensors history")
     @ApiResponses(value = {
-            @ApiResponse(code = 404, message = "Specified actuator does not exist")
+            @ApiResponse(code = 404, message = "Specified sensor does not exist")
     })
-    public SensorHistoryResponse getHistory(@PathParam("id") String id) {
-        throw new UnsupportedOperationException();
+    public SensorHistoryResponse getHistory(@PathParam("id") String id, @PathParam("dataSize") int size) {
```

Comment:
Chang dataSize to a query param, and provide defaults. maybe 50?

Reply:
So the `@ApiPath` would look like "{id}/history", and you can just label the `dataSize` as a query parameter instead